plugins/hls-latex-plugin/src/Ide/Plugin/Latex.hs

```diff
@@ -0,0 +1,558 @@
+                                                                      usepackage,
+                                                                      (&))
+import           Text.LaTeX.Base.Render                              (render)
+import           Text.LaTeX.Base.Syntax                              (LaTeX (TeXEnv),
+                                                                      MathType,
+                                                                      Measure,
+                                                                      TeXArg)
+import           Text.LaTeX.Base.Types                               (TableSpec (..))
+import           Text.LaTeX.Packages.AMSMath                         (align,
+                                                                      amsmath,
+                                                                      autoBraces,
+                                                                      autoParens,
+                                                                      autoSquareBrackets,
+                                                                      cases,
+                                                                      equation_,
+                                                                      mathrm,
+                                                                      space)
+import           Text.LaTeX.Packages.AMSSymb                         (amssymb)
+import           Text.Printf                                         (printf)
+
+data Log = LogLatex String
+  deriving Show
+
+instance Pretty Log where
+  pretty = \case
+    LogLatex log -> viaShow log
+
+
+data HaskellToLatexError = NotSupportedError String
+                         | ImageTooLargeError String
+                         | FileParseError String
+                         | NoBindingsFound String
+                         deriving Show
+
+showError :: HaskellToLatexError -> String
+showError (NotSupportedError s) = "Function contains non-supported syntax: " ++ s
+showError (ImageTooLargeError s) = "Function generates too large figure: " ++ s
+showError (FileParseError s) = "Could not parse file: " ++ s
+showError (NoBindingsFound s) = "Could not find any bindings at cursor: " ++ s
+
+deriving instance FromJSON LaTeX
+deriving instance FromJSON Measure
+deriving instance FromJSON TeXArg
+deriving instance FromJSON MathType
+
+descriptor :: Recorder (WithPriority Log) -> PluginId -> PluginDescriptor IdeState
+descriptor recorder plId =
+  (defaultPluginDescriptor plId)
+    { pluginHandlers = mkPluginHandler STextDocumentCodeAction codeActionProvider
+                    <> mkPluginHandler STextDocumentHover renderLatexOnHover
+    , pluginCommands = [PluginCommand "sendLatexNotification"
+                                      "Show Latex for this function"
+                                      sendLatexNotificationCommand
+                       ]
+    }
```

Comment:
If the plugin is flaky you should make it an "opt-in" plugin, disabled by default:

```suggestion
    { pluginHandlers = mkPluginHandler STextDocumentCodeAction codeActionProvider
                    <> mkPluginHandler STextDocumentHover renderLatexOnHover
    , pluginCommands = [PluginCommand "sendLatexNotification"
                                      "Show Latex for this function"
                                      sendLatexNotificationCommand
                       ]
     , configInitialGenericConfig = def {
        plcGlobalOn = False
      }
    }
```
Then you can enable it by setting the LSP property below in your editor config:
```
    "haskell.plugin.latex.globalOn": true,
```

When we make an HLS release, @fendor or someone else will also release the Haskell VSCode extension and your plugin will show up in the configuration pages, being easier to discover and enable

Reply:
I was also kind of hoping someone more knowledgeable than me could look into the code and see if my cursor logic to get the function and whatnot could be improved, since I kind of just went for a trial and error and settle with the first thing that worked

Reply:
Sadly we're a bit low on review cycles and it's a decent amount of code...

Reply:
That's okay, I understand, but I'd still appreciate if someone could go through the code and offer feedback. I have time, I'll just be randomly bumping this every once in a while 😛